### ISSUE TITLE
Backup to network share fails with 'The directory does not exist'

### ISSUE THREAD

**Issue by klockart** [olahallengren/sql-server-maintenance-solution#127]
We have a network share that can only be accessed using a specified domain account. We have SQL Server Agent setup to use this domain account and I have verified that the login/pw work to restart SQL Server Agent. Also, on the server where the database resides, I can create a network drive using the domain account and password.

However, when I run the backup script via SQL Agent job, I receive the error that the directory does not exist. 

I found one link that stated that the domain account cannot be longer that 20 characters and ours is 17 characters long, in the pattern DDDDDD\uuuuuuuuuu.

The error occurs around line 458 in the procedure, [master],[dbo],[DatabaseBackup], after this statement executes:
EXECUTE [master].dbo.xp_fileexist @CurrentRootDirectoryPath

When I impersonate the domain login and run the procedure, all values return 0.

We have been working with our Ops team aggressively to make sure permissions are correct.

Is there anything else that we need to check?

Thank you.

Server		XXX-X-XX-1
Job Name		Maintenance: DW_WeeklyIndexUpdStats_FullBak Sun 5:35a
Step Name		Test backup scripts SSISDB
Duration		00:00:01
Sql Severity	0
Sql Message ID	0
Operator Emailed	
Operator Net sent	
Operator Paged	
Retries Attempted	0

Message
Executed as user: DDDDDD\uuuuuuuuuu. Date and time: 2018-08-06 13:53:58  Server: XXX-X-XX-1  Version: 13.0.5149.0  Edition: Enterprise Edition: Core-based  Procedure: [master].[dbo].[DatabaseBackup]  Parameters: @Databases = 'SSISDB', 
@Directory = '\\xxx-xxxxxx\xxx-xxx-xx', @BackupType = 'FULL', @Verify = 'Y', @CleanupTime = 384, @CleanupMode = 'AFTER_BACKUP', @Compress = NULL, @CopyOnly = 'N', @ChangeBackupType = 'N', @BackupSoftware = NULL, @CheckSum = 'Y', 
@BlockSize = NULL, @BufferCount = NULL, @MaxTransferSize = NULL, @NumberOfFiles = 4, @CompressionLevel = NULL, @Description = NULL, @Threads = NULL, @Throttle = NULL, @Encrypt = 'N', @EncryptionAlgorithm = NULL, @ServerCertificate = 
NULL, @ServerAsymmetricKey = NULL, @EncryptionKey = NULL, @ReadWriteFileGroups = 'N', @OverrideBackupPreference = 'N', @NoRecovery = 'N', @URL = NULL, @Credential = NULL, @MirrorDirectory = NULL, @MirrorCleanupTime = NULL, 
@MirrorCleanupMode = 'AFTER_BACKUP', @LogToTable = 'Y', @Execute = 'Y'  Source: https://ola.hallengren.com     Msg 50000, Level 16, State 1, Server XXX-X-XX-1, Procedure DatabaseBackup, Line 458  The directory \\xxx-xxxxxx\xxx-xxx-xx
does not exist.     Msg 50000, Level 16, State 1, Server XXX-X-XX-1, Procedure DatabaseBackup, Line 741  The documentation is available at https://ola.hallengren.com/sql-server-backup.html.     Date and time: 2018-08-06 13:53:59.  
Process Exit Code 1.  The step failed.

**Comment by DiHo78** [comment]
Hi Klockart,

it is the executing SQL Server Service account which needs the permissions to Access the share. Not the executing Agent Service account.

**Comment by klockart** [comment]
Thank you for your response.

I changed the SQL Server(MSSQLSERVER) account login  from LocalSystem to the domain account, which I confirmed can access the fileshare. When I test the job, I receive the same error. I decided to take the OlaH procedure out of the equation and ran a straight 'backup database to disk'...still getting an error that I can't access the fileshare. I think the issue is our security policy, so I will continue to pursue internally. Thanks.
